### DISCUSSION TITLE
How to find if starship uses pwsh or powershell?

### DISCUSSION THREAD

**Discussion by Marin-Kitagawa** [starship/starship#4459] [Q&A] [answered]
When I execute the command `starship init pwsh | iex`, it says that `pwsh` is not supported while `powershell` is. How to find which of the shells starship uses i.e. `pwsh` (Powershell 7.2.6) or `powershell`? But I am calling the command from Windows Terminal which is running `pwsh`.

**Comment by davidkna** [comment, marked as answer]
Both older and newer versions of PowerShell (including `pwsh`)  use the same `starship init powershell` command.

**Reply by Marin-Kitagawa** [reply]
Ok. So, does that mean `starship` will use `powershell` when I call it from `powershell` and `pwsh` when called from `pwsh`? I actually viewed the source code for `starship`. There were code like `powershell | pwsh`. So, I wasn't sure what would be taken

**Reply by Marin-Kitagawa** [reply]
I also saw this
```pwsh
# /src/init/starship.ps1
if ($PSVersionTable.PSVersion.Major -gt 5) {
        $ENV:STARSHIP_SHELL = "pwsh"
} else {
        $ENV:STARSHIP_SHELL = "powershell"
}
```

This means that, the version is checked and automatically the greater version (in this case `pwsh`) is chosen.

**Reply by davidkna** [reply]
The pwsh value is only internally to use the correct shell version in custom modules. Externally, only the `powershell` value is exposed.
